src/main/java/org/apache/sysml/parser/dml/Dml.g4

```diff
@@ -152,6 +152,8 @@ expression returns [ org.apache.sysml.parser.common.ExpressionInfo info ]
     | left=expression op=('&'|'&&') right=expression # BooleanAndExpression
     // boolean or
     | left=expression op=('|'|'||') right=expression # BooleanOrExpression
+    // boolean xor
+    | op='xor' '(' left=expression ','  right=expression ')' # BooleanXorExpression
```

Comment:
I would highly recommend not adding `xor` as keyword in our grammar. Instead please consider adding xor as a builtin binary function. Most of your code will remain same, except for the language layer. You can use 
 https://github.com/apache/systemml/commit/50a895f86167d5766b249a3db89f9c2b06f7a0d0 as an example for adding a builtin at a language layer :)

Here is a sample R script that demonstrates overriding of builtin function:
```
> xor(1, 0)
[1] TRUE
> xor <- function(a, b) { a + b }
> xor(1, 0)
[1] 1
```

Here is a sample DML script that demonstrates overriding of builtin function:
```
$ cat temp.dml
max <- function(double a, double b) return (double c)  {
        c = a + b
}

d = max(1, 2)
print(d)
$ ~/spark-2.2.0-bin-hadoop2.7/bin/spark-submit SystemML.jar -f temp.dml
17/09/21 10:20:48 INFO DMLScript: BEGIN DML run 09/21/2017 10:20:48
17/09/21 10:20:48 INFO DMLConfig: Using internal default configuration settings.  If you wish to customize any settings, please supply a `SystemML-config.xml` file.
17/09/21 10:20:48 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
3.0
SystemML Statistics:
Total execution time:           0.000 sec.
Number of executed Spark inst:  0.
```